Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -367,16 +367,49 @@ public class ResponseAccumulator: HTTPClientResponseDelegate {
         case error(Error)
     }
 
+    public struct ResponseTooBigError: Error, CustomStringConvertible {
+        var maxBodySize: Int
+
+        public var description: String {
+            return "ResponseTooBigError: received response body exceeds maximum accepted size of \(self.maxBodySize) bytes"
+        }
+    }
+
     var state = State.idle
     let request: HTTPClient.Request
 
+    static let maxByteBufferSize = Int(UInt32.max)
+
+    /// Maximum size in bytes of the HTTP response body that ``ResponseAccumulator`` will accept
+    /// until it will abort the request and throw an ``ResponseTooBigError``.
+    ///
+    /// Default is 2^32.
+    /// - precondition: not allowed to exceed 2^32 because ``ByteBuffer`` can not store more bytes
+    public var maxBodySize: Int = maxByteBufferSize {
+        didSet {
+            precondition(self.maxBodySize >= 0, "maxBodyLength is not allowed to be negative")
+            precondition(
+                self.maxBodySize <= Self.maxByteBufferSize,
+                "maxBodyLength is not allowed to exceed 2^32 because ByteBuffer can not store more bytes"
+            )
+        }
+    }
+
     public init(request: HTTPClient.Request) {
         self.request = request
     }
 
     public func didReceiveHead(task: HTTPClient.Task<Response>, _ head: HTTPResponseHead) -> EventLoopFuture<Void> {
         switch self.state {
         case .idle:
+            if let contentLength = head.headers.first(name: "Content-Length"),
+               let announcedBodySize = Int(contentLength),
+               announcedBodySize > self.maxBodySize {
+                let error = ResponseTooBigError(maxBodySize: maxBodySize)
```

Comment:
We want to be a _little_ careful here: if this is a response to a HEAD request, we don't expect a response body at all so we don't need to worry about this.